infrastructure/src/main/java/org/corfudb/infrastructure/log/LogMetadata.java

```diff
@@ -74,10 +71,10 @@ public void update(LogData entry, boolean initialize, long globalTrimMark) {
                 // states of the actual address map. In the case of log updates, the trim mark will be set
                 // as prefix trims are performed.
                 if (addressSpace == null) {
-                    Long streamTrimMark = Address.NON_EXIST;
+                    long streamTrimMark = Address.NON_EXIST;
```

Comment:
If a stream is opened first time with a log write (not initialize), would it be better to use globalTrimMark for its streamTrimMark instead of -1? (Maybe less trimmedException here: https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java#L160 ? Not sure...)

Reply:
That would actually lead to more trimmedExceptions, many of them false. Think of these scenarios:

1. That stream didn't exist by the time the trim happened, hence there are no checkpoints for it. We would think the trim mark is global trim mark and fail to find a cp that covers that space.
2. Because we don't persist snapshot address in our checkpoint entries we would also fail.